build: Add meson build system

jtojnar opened the pull request:
https://wiki.gnome.org/Initiatives/GnomeGoals/MesonPorting

Kaffeine commented:
:open_mouth: . I'm sorry I know nothing about Meson and maintain mostly Qt parts. If this works for you and there is no objection from anyone then I'm agree to merge it. :confused:

jtojnar commented on .gitignore:
```diff
@@ -48,6 +48,7 @@ Makefile.in
 data/org.freedesktop.Telepathy.ConnectionManager.idle.service
 
 extensions/_gen
```
What I do not like is that the extensions are generated inside the build tree.

jtojnar replied:
Actually, they are not with meson, so everything is fine.

jtojnar commented on meson.build:
```diff
@@ -0,0 +1,87 @@
+	version: '0.2.0.1',
+	default_options: [
+		'c_std=c99',
+		# 'werror=@0@'.format((meson.project_version().split('.')[3] == '0').to_string()), # development version
```
This will need to be removed: https://github.com/mesonbuild/meson/issues/4577#issuecomment-470117483

okias commented:
@jtojnar @Kaffeine  can be merged then? Meson would be nice

Kaffeine commented:
@jtojnar please remove the WIP mark from the PR title if you're OK with this being merged.

I would be happy to merge this as soon as the author agree, I'll be unavailable until November 5.
It would be fantastic to eventually have meson build for telepathy-mission-control and telepathy-glib too.

jtojnar commented:
I dropped https://github.com/TelepathyIM/telepathy-idle/pull/4#discussion_r265544907, switched to finding Python using the [python module](https://mesonbuild.com/Python-module.html) (requires bumping minimal Meson version), moved twisted test to a separate suite so we can easily disable it, and fixed formatting.

Kaffeine commented:
Merged as 108e30c59bf2c7f96611129c67d060f232ddbaf4.
Thank you!

Kaffeine commented:
I am sorry, I messed up with your branch. You targeted `master` before Python 3 support landed and I merged the code to `telepathy-idle-0.2` after the code switched to P3. :facepalm:.
I fixed the Meson build in the 0.2.2 release (just aired).
